torchdata/datapipes/iter/load/s3io.py

```diff
@@ -40,15 +40,27 @@ class S3FileListerIterDataPipe(IterDataPipe[str]):
         region: region for access files (inferred from credentials by default)
 
     Example:
-        >>> from torchdata.datapipes.iter import IterableWrapper, S3FileLister
-        >>> s3_prefixes = IterableWrapper(['s3://bucket-name/folder/', ...])
-        >>> dp_s3_urls = S3FileLister(s3_prefixes)
-        >>> for d in dp_s3_urls:
-        ...     pass
+
+    .. testsetup::
+
+        from torchdata.datapipes.iter import IterableWrapper, S3FileLister
+
+        S3FileLister.__iter__ = lambda self: iter([])
```

Comment:
This is too hacky. How about construct an empty list as `s3_prefixes` and add a comment about the pattern of each path.

Reply:
imho if we use an empty list of files, the example is not very descriptive. `IterableWrapper(['s3://bucket-name/folder/', ...])` is necessary for the user to understand the example.

We could mock here, but we would basically do the same thing of emptying the list in order to make it work. But mocking would alter the example. So I find it difficult to achieve a good solution. Please note that the test setup is not shown in the example and that we don't really want to alter the example here.

Reply:
I'll try patching. As I cannot use a with block, I will use patch.start() in testsetup and patch.stop() in testcleanup